lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
```

Comment:
Not sure I entirely agree but this is arguing semantics at this point. `CC::Engine::Analyzers::ParserError` means anything we want. It can certainly mean "Syntax errors only", but by name, it means there was an error in the parser. Whether that is "syntax error", "took too long", "not supported", or whatever... The exception can be made to wrap another exception to provide more details if we want, tho, in practice, I've found that the CC engine generally does NOT provide more details where there are errors. That's why I rescued and logged where I did.